crates/node_types/lightclient/src/lightclient.rs

```diff
@@ -278,51 +282,69 @@ impl LightClient {
-                }
+        if finalized_epoch.public_values.len() < 64 {
+            return Err(anyhow::anyhow!(
+                "Public values length is less than 64 bytes"
+            ));
+        }
 
-                if finalized_epoch.public_values.len() < 64 {
-                    return Err(anyhow::anyhow!(
-                        "Public values length is less than 64 bytes"
-                    ));
-                }
+        // Extract and verify commitments
+        let (proof_prev_commitment, proof_current_commitment) =
+            self.extract_commitments(&finalized_epoch.public_values)?;
 
-                // Extract and verify commitments
-                let (proof_prev_commitment, proof_current_commitment) =
-                    self.extract_commitments(&finalized_epoch.public_values)?;
+        self.verify_commitments(
+            finalized_epoch,
+            proof_prev_commitment,
+            proof_current_commitment,
+        )?;
 
-                self.verify_commitments(
-                    &finalized_epoch,
-                    proof_prev_commitment,
-                    proof_current_commitment,
-                )?;
+        // Update latest commitment
+        self.latest_commitment.write().await.replace(proof_current_commitment);
 
-                // Update latest commitment
-                self.latest_commitment.write().await.replace(proof_current_commitment);
+        // Verify SNARK proof
+        self.verify_snark_proof(finalized_epoch, finalized_epoch.public_values.as_slice())?;
 
-                // Verify SNARK proof
-                self.verify_snark_proof(
-                    &finalized_epoch,
-                    finalized_epoch.public_values.as_slice(),
-                )?;
+        self.event_publisher.send(LightClientEvent::EpochVerified {
+            height: finalized_epoch.height,
+        });
 
-                self.event_publisher.send(LightClientEvent::EpochVerified {
-                    height: finalized_epoch.height,
-                });
+        Ok(())
+    }
 
-                Ok(())
-            }
-            Ok(None) => {
-                self.event_publisher.send(LightClientEvent::NoEpochFound { height });
+    async fn process_epoch(&self, height: u64) -> Result<()> {
+        info!("processing epoch at height {}", height);
+        self.event_publisher.send(LightClientEvent::EpochVerificationStarted { height });
+
+        match self.da.get_finalized_epoch(height).await {
+            Ok(finalized_epochs) => {
+                if finalized_epochs.is_empty() {
+                    self.event_publisher.send(LightClientEvent::NoEpochFound { height });
+                }
+
+                // Process each finalized epoch
+                for epoch in finalized_epochs {
+                    match self.poop_epoch(&epoch).await {
+                        Ok(()) => {
+                            self.event_publisher.send(LightClientEvent::EpochVerified {
```

Comment:
you are already doing this on line 314, so remove it

Then the match can probably be simplified, because you really only are "handling" the `Err` case